matrix_client/client.py

```diff
@@ -127,6 +127,10 @@ def start_listener_thread(self, timeout=30000):
             e = sys.exc_info()[0]
             print("Error: unable to start thread. " + str(e))
 
+    def upload(self,content,content_type):
+        response = self.api.media_upload(content,content_type)
+        return response["content_uri"]
```

Comment:
I don't think you understood my previous request on this. Let's say I'm a user of the SDK. I want to upload a file using this method and if something goes wrong **for whatever reason** I want to display an error to the user. How would you do it? Perhaps you'd do:

```python
try:
  client.upload(content, "image/jpeg")
except MatrixRequestError as e:
  print "Failed to upload: %s" % (e,)
```

Now let's say we update Synapse and tweak the returned JSON object from this upload API. `content_uri` no longer exists. This code will now *not* catch all errors because it will `KeyError` on this line. We've broken the terms of our API contract and leaked through exceptions we didn't tell the caller we'd throw. We need to check that this key exists and fail with the right error if it doesn't, e.g:

```python
try:
  return response["content_uri"]
except KeyError:
  raise MatrixRequestError(content="Cannot parse returned JSON")
```

Part of the reason for using client SDKs is to shield the caller from disruptive API changes and to present a clear and consistent API for people to use. That's why I'm making such a fuss about this :)